Date: Dec 11, 2018 1:56 AM 

From: a91e26dc7ea57c67227898c2a8cfb685795db173 Sep 20, 2019 2:46 PM

Subject: [PATCH] XONE-19886: force reboot on systemd freeze

	Added a force reboot to ensure the box reboot on systemd freeze
     
Source: COMCAST

License: LGPL-2.1-or-later

Upstream-Status: Pending

Signed-off-by: Vinu Palackal <Vinu_P@cable.comcast.com>
---

Index: git/src/core/main.c
===================================================================
--- git.orig/src/core/main.c
+++ git/src/core/main.c
@@ -129,6 +129,8 @@ static void pager_open_if_enabled(void)
 
 noreturn static void crash(int sig) {
 
+        // create a FILE typed pointer
+	FILE *file_pointer; 
         if (getpid() != 1)
                 /* Pass this on immediately, if this is not PID 1 */
                 raise(sig);
@@ -216,6 +218,15 @@ noreturn static void crash(int sig) {
         }
 
         log_info("Freezing execution.");
+        log_info("Systemd went into freeze Mode, we will do a force reboot on next reboot call");
+       	// open the file "name_of_file.txt" for writing
+	file_pointer = fopen("/tmp/systemd_freeze_reboot_on", "w"); 
+ 
+	// Write to the file
+	fprintf(file_pointer, "REBOOT");
+	
+	// Close the file
+	fclose(file_pointer);  
         freeze();
 }
 
